### PR TITLE
added broadcast to everyone else

### PR DESCRIPTION
**Pull request by dirkk0**
https://github.com/websockets/ws/issues/465

### CODE REVIEW

**Comment by lpinca** [comment]
@dirkk0 thank you for this patch.
I find the example a bit redundant as there is a broadcast to all example just above it.

What do you think about adding a sentence to the broadcast to all example explaining that if you want to exclude the sender you have to filter it out from the clients loop?

**Comment by dirkk0** [comment]
Hi, thanks for your response! Sure, we could do this also this way, but in
most of my work the exclusion is rather the default, no?
So maybe just one example, with the filter in, commented out with a comment
saying 'Uncomment this to exclude the sender'?

**Comment by lpinca** [comment]
It makes sense. 👍

**Comment by dirkk0** [comment]
ok, it's a micro patch, but may help.

File: README.md [hunk 1]
```diff
@@ -130,6 +130,8 @@ var WebSocketServer = require('ws').Server
 
 wss.broadcast = function broadcast(data) {
   wss.clients.forEach(function each(client) {
+    // use this, if you want to send to everyone else
+    // if (client !== ws) client.send(data);
```
Comment (lpinca) [inline README.md]: The only thing I'm not sure is that the reader may wonder what `ws` is as it isn't defined in this example, otherwise looks good.

Comment (dirkk0) [inline README.md]: good point.

**Comment by dirkk0** [comment]
I changed it accordingly.

**Comment by lpinca** [comment]
@dirkk0 sorry to pester but what do you think about changing it like this?
```js
var WebSocketServer = require('ws').Server
  , wss = new WebSocketServer({ port: 8080 });

// Broadcast to all.
wss.broadcast = function broadcast(data) {
  wss.clients.forEach(function each(client) {
    client.send(data);
  });
};

wss.on('connection', function connection(ws) {
  ws.on('message', function message(data) {
    // Broadcast to to everyone else.
    wss.clients.forEach(function each(client) {
      if (client !== ws) client.send(data);
    });
  })
});
```

**Comment by dirkk0** [comment]
You are not pestering me at all -  it's me who is wasting your time! :-)
This is a great solution, please go for it.
Thanks!

**Comment by lpinca** [comment]
Update the PR when you have time, so we can this merged :)
Thanks.

**Comment by lpinca** [comment]
Merged as 78425d0c3360acd67280f33cab89c84a46d3eb0d.
I hope you don't mind not having "purple merged status" on this PR.
Thanks!

**Comment by dirkk0** [comment]
not at all, thx!

**Pull request closed without merging**